core-v1/src/main/scala/algebra/Expr.scala

```diff
@@ -543,12 +546,15 @@ object Expr {
     *
     * @param lens a lens from the input type to some selected field
     */
-  final case class Select[-I, +O : OP, OP[_]](
-    lens: VariantLens[I, O],
+  final case class Select[W[+_] : Extract, -A, +B, +O : OP, OP[_]](
+    lens: VariantLens[A, B],
     override private[v1] val debugging: Debugging[Nothing, Nothing] = NoDebugging,
-  ) extends Expr[I, O, OP]("select") {
-    override def visit[G[-_, +_]](v: Visitor[G, OP]): G[I, O] = v.visitSelect(this)
-    override private[v1] def withDebugging(debugging: Debugging[Nothing, Nothing]): Select[I, O, OP] =
+  )(implicit
+    ot: SelectFromWrapped.Aux[W, B, O],
+    // TODO: Is it worth it to avoid wrapping this input? Maybe take an input expression?
+  ) extends Expr[W[A], O, OP]("select") {
+    override def visit[G[-_, +_]](v: Visitor[G, OP]): G[W[A], O] = v.visitSelect(this)
```

Comment:
The benefit of using a Select node that takes in the input expression is that we can attach a debugger to the see the state of the selected value before wrapping the elements... might be worth it.

Reply:
I updated this PR to add more types... it's a bit uglier now, but you can debug the intermediate values... not sure if it is too much.